# Langkah #1 - Analisa Descriptive Statistic
---

In [1]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [2]:
df = pd.read_csv('insurance.csv')
df.head()

age     sex     bmi  children smoker     region      charges
0   19  female  27.900         0    yes  southwest  16884.92400
1   18    male  33.770         1     no  southeast   1725.55230
2   28    male  33.000         3     no  southeast   4449.46200
3   33    male  22.705         0     no  northwest  21984.47061
4   32    male  28.880         0     no  northwest   3866.85520

---
**6. Apakah rata rata umur perempuan dan laki-laki yang merokok sama?**

In [3]:
gender_smoke = df[['age', 'sex', 'smoker']].where(df['smoker']=='yes')
np.round(gender_smoke['age'].groupby(gender_smoke['sex']).mean(), 2)

sex
female    38.61
male      38.45
Name: age, dtype: float64

Jadi, secara umum rata-rata umur laki-laki dan perempuan yang merokok tidak jauh berbeda.

---
**7. Mana yang lebih tinggi, rata rata tagihan kesehatan perokok atau non merokok?**

In [4]:
charges_smoker = df[['charges', 'smoker']].groupby(['smoker']).mean()
np.round(charges_smoker, 2)

charges
smoker          
no       8434.27
yes     32050.23

Jadi, jumlah tagihan orang merokok lebih tinggi daripada yang tidak merokok yaitu sebesar 32050.23.

---
**8. Mana yang lebih tinggi, rata rata tagihan kesehatan perokok yang BMI nya diatas 25 atau non perokok yang BMI nya diatas 25**

In [5]:

bmi_smoker = df[['charges', 'bmi', 'smoker']].where(df['bmi']>25)
bmi_smoker.head()

charges    bmi smoker
0  16884.9240  27.90    yes
1   1725.5523  33.77     no
2   4449.4620  33.00     no
3         NaN    NaN    NaN
4   3866.8552  28.88     no

In [6]:
np.round(bmi_smoker['charges'].groupby(bmi_smoker['smoker']).mean(), 2)

smoker
no      8629.59
yes    35116.91
Name: charges, dtype: float64

Jadi, rata-rata tagihan kesehatan yang BMI nya di atas 25 adalah yang perokok yaitu sebesar 35116.91.

---
**9. BMI mana yang lebih tinggi, seseorang laki-laki atau perempuan?**

In [7]:
np.round(df[['bmi', 'sex']].groupby(['sex'])['bmi'].mean(), 2)

sex
female    30.38
male      30.94
Name: bmi, dtype: float64

Jadi, secara rata-rata BMI antara kedua gender tidak terpaut jauh, namun yang tertinggi adalah laki-laki yaitu sebesar 30.94.

---
**10. BMI mana yang lebih tinggi, seseorang perokok atau non perokok?**

In [8]:
np.round(df[['bmi', 'smoker']].groupby(['smoker'])['bmi'].mean(), 2)

smoker
no     30.65
yes    30.71
Name: bmi, dtype: float64

BMI antara perokok ataupun non perokok tidak terlalu jauh, namun yang tertinggi sebesar 30.71 yaitu dari yang perokok.